files/private-chef-ctl-commands/helpers/key_ctl_helper.rb

```diff
@@ -0,0 +1,165 @@
+require "chef/config"
+require "chef/rest"
+
+class KeyCtlHelper
+  def initialize
+    @chef_rest = configure_chef_rest
+  end
+
+  # Optparse doesn't properly handle the case where you specify an argument with mandatory input
+  # and then pass another argument after it. It detects the second argument as input to the first argument.
+  #
+  # For example, if you have two arguments with mandatory input, --username NAME and
+  # --public-key-path PATH, defined in optparse, and you pass:
+  #
+  # cmd --username --public-key-path /etc/key
+  #
+  # It will detect the input to --username as --public-key-path and won't even detect --public-key-path as an argument.
```

Comment:
Just curious, what version of ruby were you using to test this? I seem to remember this being a bug and everything using opt parse (mixlib-cli is based on it) could not take arguments beginning with a hyphen.